routing/router.go

```diff
@@ -1749,36 +1750,97 @@ func outgoingFromIncoming(incomingAmt lnwire.MilliSatoshi,
 	// In case the inbound fee rate is less than -1 (-100%), we fail to
 	// compute the outbound amount and return the incoming amount. This also
 	// protects against zero division later.
-	m := (PPM*PPM + Ri*PPM + Ro*PPM + Ro*Ri) / (PPM * PPM)
-	if m <= 0 {
+	// m := (PPM*PPM + Ri*PPM + Ro*PPM + Ro*Ri) / (PPM * PPM)
+
+	// Note: this could maybe fit into an int64.
+	m1 := new(big.Int).Mul(PPM, PPM)
```

Comment:
I think this reads fine. However if you *want* to improve it you can always postfix these lines with their more familiar expressions like this:

```golang
m1 := new(big.Int).Mul(PPM, PPM) // m1 := PPM * PPM
...
// m := m1 + m2 + m3 + m4
m := big.NewInt(0)
m.Add(m, m1)
...
m.Add(m, m4)
```

Reply:
I added some comments to make the calculations hopefully easier to follow